docs/language/builtins.rst

```diff
@@ -346,6 +346,28 @@ is_contract(address AccountId) returns (bool)
 
 Only available on Substrate. Checks whether the given address is a contract address. 
 
+set_code_hash(bytes hash) returns (uint32)
+++++++++++++++++++++++++++++++++++++++++++
+
+Only available on Substrate. Replace the contracts code with the code corresponding to ``hash``.
+Implies that the new code was already uploaded, otherwise the operation fails.
+A return value of 0 indicates success; a return value of 7 indicates that there was no corresponding code found.
+
+.. note::
+
+    This is a low level function. We strongly advise consulting the underlying 
+    `API documentation <https://docs.rs/pallet-contracts/latest/pallet_contracts/api_doc/trait.Version0.html#tymethod.set_code_hash>`_ 
+    to obtain a full understanding of its implications.
+
+This functionality is intended to be used for implementing upgradeable contracts. 
+Pitfalls generally applying to writing
+`upgradeable contracts <https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable>`_ 
+must be considered whenever using this builtin function, most notably:
+
+* The contract must safeguard access to this functionality, so that it is only callable by priviledged users.
+* The code you are upgrading to must be storage compatible with the existing code.
```

Comment:
Isn't it clear to say that the contract's storage layout should be compatible with the new code?
Alternatively, a few words on `storage compatible` here would help understanding.

Reply:
I think saying that the new _code_ must be storage compatible is technically more correct than just saying _contract_. The _contract_ will stay at the same address. `set_code_hash` doesn't change the contract, but merely its _code_.

However I agree that we could state what this exactly means.I added a link to the OpenZepplin docs.